enginetest/query_plans.go

```diff
@@ -2033,6 +2029,119 @@ var PlanTests = []QueryPlanTest{
 			" └─ IndexedTableAccess(one_pk on [one_pk.pk] with ranges: [{[1, 1]}])\n" +
 			"",
 	},
+	{
+		Query: `SELECT a.* FROM one_pk a CROSS JOIN one_pk c LEFT JOIN one_pk b ON b.pk = c.pk and b.pk = a.pk`,
+		ExpectedPlan: "Project(a.pk, a.c1, a.c2, a.c3, a.c4, a.c5)\n" +
+			" └─ LeftIndexedJoin((b.pk = c.pk) AND (b.pk = a.pk))\n" +
+			"     ├─ CrossJoin\n" +
+			"     │   ├─ TableAlias(a)\n" +
+			"     │   │   └─ Table(one_pk)\n" +
+			"     │   └─ TableAlias(c)\n" +
+			"     │       └─ Table(one_pk)\n" +
+			"     └─ TableAlias(b)\n" +
+			"         └─ IndexedTableAccess(one_pk on [one_pk.pk])\n" +
+			"",
+	},
+	{
+		Query: `SELECT a.* FROM one_pk a CROSS JOIN one_pk c RIGHT JOIN one_pk b ON b.pk = c.pk and b.pk = a.pk`,
+		ExpectedPlan: "Project(a.pk, a.c1, a.c2, a.c3, a.c4, a.c5)\n" +
+			" └─ LeftIndexedJoin((b.pk = c.pk) AND (b.pk = a.pk))\n" +
```

Comment:
This would appear to reverse the meaning of the RIGHT JOIN. I don't mean the fact that it's a LeftIndexedJoin, I mean the fact that `b` is the secondary table here instead of the primary. `b` has to be the primary table, you can't do an index lookup on it.

Reply:
You are correct that our execution machinery breaks down with this plan. My understanding is the primary table should be the cross join, for a secondary lookup into an indexed table. I added a dozen more tests to verify our query join correctness, and disabled the optimization specifically for RIGHT_JOIN in the meantime. I am lacking some understanding of both index generation and the execution machinery for applying those to right joins, but otherwise this PR should work for LEFT_JOIN with CROSS_JOINs.